crates/goose/src/session/storage.rs

```diff
@@ -358,9 +335,7 @@ pub async fn generate_description(
     metadata.description = description;
 
     // Update the file with the new metadata and existing messages
-    update_metadata(session_file, &metadata).await?;
-
-    Ok(())
+    return save_messages_with_metadata(session_file, &metadata, messages);
```

Comment:
unrelated rust noob question, why do I need the explicit return statement here? `update_metadata` has the same return type but doesn't have an explicit return, even though the last line is a call to `save_messages_with_metadata`, just like here (line 347). without this return, I get ` expected enum `Result<(), anyhow::Error>`found unit type `()``

Reply:
Nvm, I figured it out, it's the semicolon! I guess that makes sense.